addons/website_slides/models/slide_channel.py

```diff
@@ -760,7 +758,24 @@ def _remove_membership(self, partner_ids):
         self.message_unsubscribe(partner_ids=partner_ids)
 
         if removed_channel_partner_domain:
-            self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain).unlink()
+            removed_channel_partner = self.env['slide.channel.partner'].sudo().search(removed_channel_partner_domain)
+            if removed_channel_partner:
+                removed_channel_partner.action_archive()
+
+    def _update_users_karma(self, partner_ids, joining=True):
```

Comment:
@awa-odoo I think now we don't need this method `_update_users_karma` anymore because removing the membership should NOT subtract the karma right? So afterward while re-joining, karma will not get affected and there is nothing to do with it. What's say? If I am wrong then correct me.

Reply:
You are right, leaving should not affect karma anymore.  So you can indeed clean this method.